### PR TITLE
Test gulp styles task

### PR DESCRIPTION
**Pull request by ataylorme**
## Description

Follow up of #395. Adds Jest to test gulp styles task for expected outcomes based on different configuration settings. The tests can be ran with `npm run test-gulp`.

## List of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] This pull request relates to a ticket.
- [x] This code is tested.
- [ ] This change has been added to CHANGELOG.md
- [x] I want my code added to WP Rig.

### CODE REVIEW

**Comment by ataylorme** [comment]
@phated I thought you might be curious about the progress on this. See [my comment in #279](https://github.com/wprig/wprig/issues/279#issuecomment-484581345). I was able to use a TDD workflow to adjust the gulp process to allow configuration of `autoprefixer`. It was a _much_ better workflow than testing the expected gulp output manually. Thanks again for helping me in gulp office hours!

**Comment by phated** [comment]
@ataylorme that's awesome!  great work!

**Comment by mor10** [comment]
Can you give a real-world example of how this would be used?

**Comment by ataylorme** [comment]
> Can you give a real-world example of how this would be used?

@mor10 I am not sure what you mean. This is for internal use to tests the gulp tasks as we develop WP Rig, rather than testing the expected results manually.

I do not expect WP Rig end users to run these tests.

**Comment by ataylorme** [comment]
@mor10 turns out Node was already on Travis for `eslint`. I was able to add the gulp tests to Travis fairly easily. [The tests passed](https://travis-ci.com/wprig/wprig/jobs/195654417) to boot.

File: gulp/tests/styles.test.js [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+import { getThemeConfig, getDefaultConfig } from '../utils';
+import { stylesAfterReplacementStream } from '../styles';
+
+function makeMockFiles() {
+  return [
+    new Vinyl({
+      path: 'mock.css',
+      contents: Buffer.from(
+`
+@import "_custom-properties.css";
```
Comment (bamadesigner) [inline gulp/tests/styles.test.js]: Can we fix the tabbing here?

**Pull request merged**